File: JSPM.Web/Modules/Master/Skills/SkillsRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace JSPM.Master;

[ConnectionKey("Default"), Module("Master"), TableName("[Master].[Skills]")]
[DisplayName("Skills"), InstanceName("Skills")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
public sealed class SkillsRow : Row<SkillsRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Identity, IdProperty]
    public int? Id
    {
        get => fields.Id[this];
        set => fields.Id[this] = value;
    }

    [DisplayName("Name"), Size(100), NotNull, QuickSearch, NameProperty]
    public string Name
    {
        get => fields.Name[this];
        set => fields.Name[this] = value;
    }

    public class RowFields : RowFieldsBase
    {
        public Int32Field Id;
        public StringField Name;

    }
}

File: JSPM.Web/Modules/ServerTypes/Master/SkillsRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface SkillsRow {
    Id?: number;
    Name?: string;
}

export abstract class SkillsRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'Master.Skills';
    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<SkillsRow>();
}